### PR TITLE
feat: build script check for go and node versions

### PR DESCRIPTION
**Pull request by eliziario**
## Description:
Makes scripts/build.sh checked for specific go and node versions instead of letting build fail later with obscure error during for example, yarn install.  

## Is this change user facing?

* [ ] Yes
* [X] No

## References (if applicable):
[ISSUE: 239](https://github.com/kurtosis-tech/kurtosis/issues/239)

### CODE REVIEW

**Comment by eliziario** [comment]
@kurtosis-tech/engineers It looks like circle ci is refusing to run the pipeline on my PR :-)

**Comment by mieubrisse** [comment]
Hey @eliziario , thanks for submitting this and congratulations on being our first external contributor!!! @victorcolombo is going to review shortly.

Also, if I'm correctly interpreting your name and you're Brazilian - bem-vindo ao Kurtosis! @victorcolombo é brasileiro, e @galenmarchetti e eu somos americanos apaixonados pelo Brasil e morando aqui :)

**Comment by eliziario** [comment]
Thanks @mieubrisse! Yes, I am a brazilian and carioca. Tudo beleza ai @victorcolombo and @galenmarchetti? :-)

File: scripts/versions_check.sh [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+  if ! nvm list "${NODE_VERSION}" &> /dev/null; then
+    echo "${RED_BG}${WHITE_FG}${BOLD}node "${NODE_VERSION}" not installed. Please install it with ${NORMAL_BG}"
+    echo "${RED_BG}${WHITE_FG}nvm install "${NODE_VERSION}"                                ${NORMAL_BG}"
+    echo  ""
+    error=true
+  fi
+}
+
+check_go_version() {
```
Comment (victorcolombo) [inline scripts/versions_check.sh:58]: For Node, I think it's ok to enforce the pinned version for full consistency, given that we are using `NVM` as a version manager, allowing people to have multiple Node versions on their machine.

But in Go's case, we are not (yet) using any version manager, and we would like to allow folks with Go `1.19` and forward to also be able to build Kurtosis, but right now this errors out. Could you add support for this in the function?

**Review by victorcolombo — CHANGES_REQUESTED** [review]
Hey @eliziario ! Thanks for the first contribution on this repo 😃 

Great job on the instructions to install the correct node version, this will help a lot of folks trying to build Kurtosis from scratch.

I left one comment, once that is addressed we are good to merge!

**Comment by victorcolombo** [comment]
Additionally I see there is some CI weirdness going on (we might have a misconfigured permissions), I'm working on it right now to address it

File: scripts/versions_check.sh (hunk 1 above)
Comment (eliziario) [reply scripts/versions_check.sh:58]: I agree, Go is very well-behaved in terms of not introducing break changes, so there is no real benefit in pinning the version, only ensuring the minimal version. I will do this change later today and push it.

**Review by victorcolombo — APPROVED** [review]
Looks good to me @eliziario! I went ahead and approved it, will work on merging this and will let you know once this is released 🚀

**Comment by victorcolombo** [comment]
@eliziario Sua mudança já tá no release `0.69.2`! 🇧🇷

**Comment by mieubrisse** [comment]
@eliziario congrats on being the first external contribution to Kurtosis!! I'd love to learn more about what you're thinking of using Kurtosis for; would you be down to schedule a chat with me? You can do so with `kurtosis feedback --calendly` if that sounds good!

**Pull request merged**